pytorch_lightning/trainer/trainer.py

```diff
@@ -713,17 +713,64 @@ def tng_tqdm_dic(self):
     # -----------------------------
     # MODEL TRAINING
     # -----------------------------
-    def fit(self, model):
+    def fit(self, model, train_dataloader=None, val_dataloader=None, test_dataloader=None):
         r"""
         Runs the full optimization routine.
 
-        Example::
+        Args:
+            model (LightningModule): Model to fit.
+            Example::
 
-            trainer = Trainer()
-            model = LightningModule()
+                trainer = Trainer()
+                model = LightningModule()
+                trainer.fit(model)
+
+            train_dataloader (:class:`.torch.utils.data.DataLoader`): A Pytorch
+                DataLoader with training samples. If the model has
+                a predefined train_dataloader method this will be skipped.
+
+            val_dataloader (:class:`.torch.utils.data.DataLoader`): A Pytorch
+                DataLoader with validation samples. If the model has
+                a predefined val_dataloader method this will be skipped
+
+            test_dataloader (:class:`.torch.utils.data.DataLoader`): A Pytorch
+                DataLoader with test samples. If the model has a
+                a predefined test_dataloader method this will be skipped
 
-            trainer.fit()
         """
+        if train_dataloader:
```

Comment:
I would make it a function since the three are almost the same:
```
def _set_dataloader(dataloader, attribute):
    if isinstance(dataloader, torch.utils.data.DataLoader):
        if getattr(model, attribute) is None:
            setattr(model, attribute, lambda: dataloader)
        else:
            logging.info(f'Model has predefined `{attribute}`, '
                         'will skip the `{attribute}` passed to fit method.')
    elif dataloader:
        raise ValueError(f'`{attribute}` needs to be an instance of `torch.utils.data.DataLoader`')
    
_set_dataloader(train_dataloader, 'train_dataloader')
...
```